nifi-nar-bundles/nifi-cdc/nifi-cdc-mysql-bundle/nifi-cdc-mysql-processors/src/main/java/org/apache/nifi/cdc/mysql/processors/CaptureChangeMySQL.java

```diff
@@ -839,19 +802,18 @@ protected void connect(List<InetSocketAddress> hosts, String username, String pa
         InetSocketAddress connectedHost = null;
         Exception lastConnectException = new Exception("Unknown connection error");
 
-        if (createEnrichmentConnection) {
-            try {
-                // Ensure driverLocation and driverName are correct before establishing binlog connection
-                // to avoid failing after binlog messages are received.
-                // Actual JDBC connection is created after binlog client gets started, because we need
-                // the connect-able host same as the binlog client.
-                registerDriver(driverLocation, driverName);
-            } catch (InitializationException e) {
-                throw new RuntimeException("Failed to register JDBC driver. Ensure MySQL Driver Location(s)" +
-                        " and MySQL Driver Class Name are configured correctly. " + e, e);
-            }
+        try {
```

Comment:
Can you explain why `createEnrichmentConnection` check is no longer needed? I don't follow.

Reply:
We stopped using the DistributedMapCache and are always using an internal Map now, so we always create an enrichment connection, so we don't need to see if we need to.